src/basekernels/matern.jl

```diff
@@ -108,7 +108,7 @@ end
 
 Matern52Kernel(; metric=Euclidean()) = Matern52Kernel(metric)
 
-kappa(::Matern52Kernel, d::Real) = (1 + sqrt(5) * d + 5 * d^2 / 3) * exp(-sqrt(5) * d)
+kappa(::Matern52Kernel, d::T) where {T<:Real} = (1 + T(sqrt(5)) * d + 5 * d^2 / 3) * exp(-T(sqrt(5)) * d)
```

Comment:
Same here with sqrt(5)

Reply:
Unfortunately, IrrationalConstants doesn't define `sqrt5` (maybe should be added?) but one could still improve it a bit: ```suggestion
function kappa(::Matern52Kernel, d::Real)
    x = sqrt(oftype(d, 5))
    return (1 + x + 5 * d^2 / 3) * exp(-x)
end
```
This would also be a bit safer as it also works if `d` is not a floating point number.